### ISSUE TITLE
processNetCommand error

### ISSUE THREAD

**Issue by vinhdizzo** [powlo/script.module.pydevd#5]
Hi,

I'm trying to use this plugin on xbmc 13 (ubuntu) to debug an addon using eclipse (on windows) per [these](http://kodi.wiki/view/How-to:Debug_Python_Scripts_with_Eclipse) instructions.

I'm getting the following error in the debug console when I start the addon:

```
Debug Server at port: 5678
Traceback (most recent call last):
  File "/home/user/.xbmc/addons/script.module.pydevd/lib/pydevd.py", line 878, in processNetCommand
    thread_id, frame_id, console_command = text.split('\t', 2)
ValueError: need more than 1 value to unpack
```

Assuming this is an error with pydevd on xbmc.  Any thoughts on how I could resolve it?  Thank you.

**Comment by powlo** [comment]
Is that all you have in your stack trace? How are you initiating the debugger? Is "13" the XBMC version or the Ubuntu version?

**Comment by vinhdizzo** [comment]
Yes, that is all in the debuggibg console of eclipse.

I initiated debugger by installing the pydevd addon zip 3.4.1 i think), modifying `default.py` of the addon of interest per [these](http://kodi.wiki/view/How-to:Debug_Python_Scripts_with_Eclipse) (section 3.5; modified to `import pydevd` and specifying my eclipse ip in settrace). Then clicked on the addon in xbmc.

I am using xbmc 13 gotham on ubuntu 14.04.

**Comment by powlo** [comment]
OK thanks for the info. What command are you using in your script to launch the debugger?

**Comment by vinhdizzo** [comment]
`pydevd.settrace('eclipse ip', stdoutToServer=True, stderrToServer=True)`

**Comment by powlo** [comment]
'eclipse ip' isn't right. You need a real IP address in there. Such as '192.168.0.10'.

**Comment by vinhdizzo** [comment]
Yes I know. I put in the actual ip. I know this line works, hence the error message that displays in the original post.

**Comment by powlo** [comment]
It's possible that there's an inconsistency with the calls between Eclipse and Kodi. Ie they changed the API.

What version of Eclipse are you using, and more importantly, what version of PyDev are you running in Eclipse?

You could try manually copying the files as detailed in [3.4.1](http://kodi.wiki/view/How-to:Debug_Python_Scripts_with_Eclipse#Identify_where_PyDev_remote_debugger_Python_files_are). I'm fairly sure that will work.

**Comment by vinhdizzo** [comment]
It makes sense that pydev on eclipse 4.5.2 (com.python.pydev.debug_5.0.0.201605051159) might be incompatible with the pydevd addon on xbmc (version 3.4.1).

Are you suggesting that I copy pydev from eclipse to xbmc, or pydev from xbmc to eclipse?  I did not attempt the former because [this](http://kodi.wiki/view/How-to:Debug_Python_Scripts_with_Eclipse#Option_4:_.28Suggested_for_Linux_users.29) suggested the copying method does not work and to use the pydevd addon instead.  If so, I assume you meant the latter, so copy the old pydev source from xbmc and use that in eclipse?  Please confirm, thanks!

**Comment by vinhdizzo** [comment]
I just visited the project's github readme page, and noticed this:

```
<!-- addon.xml -->
<addon id="your.addon">
    <requires>
        <import addon="script.module.pydevd" version="3.4.1"/>
    </requires>
</addon>
```

Do I need to add this to the addon of interest?

**Comment by vinhdizzo** [comment]
Does this prove anything?  I don't see `pysrc` in eclipse's `org.python.pydev.debug_5.0.0.201605051159`.  Likewise with core.  There are appears to be a `pydev-debug.jar` file.  Odd?

**Comment by powlo** [comment]
It proves that the versions are different. I'll bet that between 3.4.1 and 4.5.2 they've changed the messaging format that takes place between the client and Eclipse.

I actually wrote another debug program that downloads and strips out the required folder: https://github.com/powlo/script.pydev.debug.

I then wrote this one (script.module.pydevd) because it was less intrusive. Maybe I should revisit the other version... add some download urls so the user can select which PyDev version they need.

I remember this was a common problem with XBMC/Kodi. It doesn't have a good python dependency manager.

So did you try copying the contents of the Eclipse pydevd folder to verify that it worked?

**Comment by vinhdizzo** [comment]
Regarding your last question, I'm still confused as to what needs to be copied and from where to where. Can you clarify? Thanks.

**Comment by powlo** [comment]
Oh you said you couldn't find anything in pysrc. Strange. I'll take a look myself tomorrow.

**Comment by vinhdizzo** [comment]
OK, I found pysrc at `org.python.pydev_5.0.0.201605051159\pysrc`, not in `org.python.debug` as the wiki suggested.

Now, where should I copy this onto ubuntu with xbmc?  As I previously mentioned, [this](http://kodi.wiki/view/How-to:Debug_Python_Scripts_with_Eclipse#Option_4:_.28Suggested_for_Linux_users.29) suggested that copying `pysrc` won't work on Linux.  Should I put `pysrc` in `~/xbmc/system/python/Lib` (nothing in `system` currently) or `/usr/share/xbmc/system/python/Lib` (only a readme.txt file in `python`?

**Comment by vinhdizzo** [comment]
I googled "xbmc ubuntu python path" and found [this](http://forum.kodi.tv/showthread.php?tid=160528) thread that you started.  I then placed the contents of `pysrc` (zipped from windows, `scp` to ubuntu, and unzipped) into `~/xbmc/addons/script.module.pydevd/lib` (basically, deleted the old content and replaced with eclipse's version of `pysrc`).  When I start the addon of interest, I get a "Working" spinning wheel.  On the eclipse side, I don't get any messages.  This tells me `pydevd.settrrace` is running (no errors, hanging), but a connection is not being established.

I think my trial and error can go to here.  Will need further hand holding in order to resume.  Thanks!

**Comment by powlo** [comment]
That sounds good. In Eclipse you have to go to the debug perspective, and in the top bar you should see some continue/step/step-out icons. You'll also see a debug connection in the left panel. Eclipse will highlight the line that it has started tracing on. More details [here.](http://www.pydev.org/manual_adv_remote_debugger.html) It might be worth just running and debugging a trivial stand alone python script, just so you minimise the variables, so to speak.

**Comment by vinhdizzo** [comment]
Ahh, will test that in a bit.  Thanks!

OT question for you: for the addon of interest, do I need to have a copy of it on xbmc (ubuntu), and a copy in eclipse's workspace?  My impression from [section 3](http://kodi.wiki/view/How-to:Debug_Python_Scripts_with_Eclipse#Create_a_new_PyDev_Project) is that a copy needs to be on both systems in order for debugging to occur (stepping through different parts of a program).  Or does it only need to be on xbmc, and eclipse will just display the relevant code (copied from xbmc)?  Thanks again.

**Comment by powlo** [comment]
My understanding is that yes you'll need a copy of it in Eclipse. Although I've never tested it. I assume that somehow Eclipse/PyDev finds a way to match the code under debug with the code in the IDE. If it can't find a match then it just goes silent, from what I recall.

FYI I noticed it's possible to install pydevd through pip: https://pypi.python.org/pypi/pydevd/0.0.5

**Comment by zbyna** [comment]
Hi,
thanks for the great addon !  After 4 days experimenting with LiClipse, PyDev and various setting according to:
http://kodi.wiki/view/How-to:Debug_Python_Scripts_with_Eclipse
I have to say that remote debugging using your addon is the only way which works on my configuration
(win7 64bit., LiClipse   3.1.0.20160817,     PyDev for Eclipse  5.2.0.20160817) 

I have encountered this error too:

```
21:57:49 T:8048   ERROR: Traceback (most recent call last):
21:57:49 T:8048   ERROR:   File "C:\Users\decak\AppData\Roaming\Kodi\addons\script.module.pydevd-master\lib\pydevd.py", line 878, in processNetCommand
21:57:49 T:8048   ERROR:     thread_id, frame_id, console_command = text.split('\t', 2)
21:57:49 T:8048   ERROR: ValueError: need more than 1 value to unpack
```

```
21:52:16 T:776  NOTICE: Starting Kodi (16.1 Git:20160424-c327c53). Platform: Windows NT x86 32-bit
21:52:16 T:776  NOTICE: Using Release Kodi x32 build
21:52:16 T:776  NOTICE: Kodi compiled Apr 24 2016 by MSVC 180040629 for Windows NT x86 32-bit version 6.0 (0x06000000)
21:52:16 T:776  NOTICE: Running on MSI MS-7681 with Windows 7 SP1,
```

**Comment by zbyna** [comment]
My problems  solved using pull request #3
It should be merged for sure.
